app/src/main/java/space/taran/arkmemo/data/repositories/TextNotesRepository.kt

```diff
@@ -17,69 +20,156 @@ import kotlin.io.path.getLastModifiedTime
 
 class TextNotesRepository @Inject constructor() {
 
-    @Inject @ApplicationContext lateinit var context: Context
+    @Inject
+    @ApplicationContext
+    lateinit var context: Context
 
-    fun saveNote(note: TextNote?) {
+    fun saveNote(note: TextNote?, rootResourceId: ResourceId? = null): Long {
+        val versionStorage = VersionStorage(getPath()!!)
         if (note != null) {
             val path = getPath()
             if (path != null) {
                 Files.list(path)
-                createTextNoteFile(
-                    path,
-                    JsonParser.parseNoteToJson(note.content),
-                )
+                val newResourceId =
+                    createTextNoteFile(
+                        path,
+                        JsonParser.parseNoteContentToJson(note.content),
+                    )!!
+                if (newResourceId < 0) {//Error creating TextNote file:
+                    return newResourceId
+                }
+                versionStorage.addVersion(newResourceId, rootResourceId)
+                return newResourceId
             }
         }
+        return CODES_CREATING_NOTE.NOTE_NOT_CREATED.errCode.toLong()
     }
 
-    fun deleteNote(note: TextNote) {
-        val filePath = getPath()?.resolve("${note.meta?.name}")
-        removeFileFromMemory(filePath)
-        Log.d("Deleted", note.meta?.name!!)
+    fun deleteNote(note: TextNote, version: Version) {
+        val notePath = getPath()?.resolve("${note.meta?.name}")
+        val versionPath =
+            getPath()?.resolve(ARK_VERSIONS_DIR)?.resolve("${version.meta?.rootResourceId}")
+        removeFileFromMemory(notePath)
+        removeFileFromMemory(versionPath)
+        //we should delete all notes listed in version that don't have been forked? by now it just deletes the last note.
```

Comment:
Good question. IMO we should delete all versions, if user initiates deletion from the main screen. _When we'll have forks, we'll need to find where the fork occurs and delete only chain of versions down to the fork, e.g. if we have versions A -> B -> C ->D and B -> E, and the user deletes D from main screen, then we remove everything except A, B and E._

If the user delete a version from versions screen, then only particular version must be removed and it's parent and children must be attached (grandparent becomes just parent): with versions A -> B -> C, deletion of B must result in versions A -> C. _When we'll have forks, all children should be lifted: with versions A -> B -> C and B -> D, deletion of B would result in A -> C and A -> D._